atomic_reactor/koji_util.py

```diff
@@ -75,8 +75,8 @@ def koji_login(session,
         # Use Kerberos
         logger.info("Using Kerberos for Koji authentication")
         if krb_principal and krb_keytab:
-            kwargs['principal'] = krb_principal
-            kwargs['keytab'] = krb_keytab
+            kwargs['principal'] = str(krb_principal)
```

Comment:
I'm assuming this is because the underlying kerberos c api doesn't like unicode?
If so, can we use something like [six](https://pythonhosted.org/six/#binary-and-text-data) to handle the conversion in both python 2 and 3?

Also, we can do the same to `hub_url` parameter in `create_koji_session`, hiding this from the plugins.

Reply:
Yes, it appears the krb c api doesn't like unicode. We could do something like six but what's the concern about python2 to python3 here? Isn't `str()` valid in both?

I don't have a real preference on how it's handled, if you'd prefer `hub_url` be a param in `create_koji_session` then :+1:

Reply:
@maxamillion The last point I had to fix was exactly the `hub_url` in `create_koji_session` :).

Reply:
If `str` works for both python versions, let's just go with that. But I don't think it will on python3. Since str there means unicode...

Reply:
@lcarva nvm, I see what you mean. It's not that it would/wouldn't work syntactically for py2/py3 but the issue of the krb5 libs not liking unicode. I'll have to find a way to test this.

Reply:
Sorry for the late reply, but there's two points of update on this now:
- The code that was broken was the krbV codepaths (which were required for keytab use), and that just does not exist under py3. The GSSAPI codepaths are totally fine with getting unicode.
- As of Koji 1.15.0, the GSSAPI codepaths are also used for keytab auth, which are happy with unicode, which makes this patch no longer needed for Fedora. The patch might still be useful for setups without gssapi (thus only krbV support), which only works with py2 anyway. And as said, GSSAPI is not picky whether it gets basestring or unicode.

Reply:
Fedora was the only one affected by this. If that's no longer true, let's drop this PR. Agree?

Reply:
Agreed. I was thinking there might've been other people, but those are likely still on Py2.